Lib/test/test_eagerly_import.py

```diff
@@ -0,0 +1,82 @@
+'''
+- Target:
+    Add a new sample test that checks Python imports are executed eagerly by default.
+
+- Description:
+    We need to confirm one of these two options based on the scenario. Otherwise, this test fails.
+        - If it is in default, the imports are executed eagerly.
+        - If Lazy Imports is enabled, the imports are NOT executed eagerly.
+'''
+
+import os
+import sys
+import unittest
+# import json
+import sqlite3
+
+class LazyImportsEnabledChecker:
+    def __init__(self):
+        self.enabled = self.check_lazy_imports()
+
+    # confirm if Lazy Imports is enabled or not
+    def check_lazy_imports(self):
+        # lazy imports is not enabled in default
+        enabled = False
+
+        # check if Lazy Imports is turned on
+        enabled |= self.check_flag()
+        enabled |= self.check_env_var()
+        return enabled
+
+    # check if the env variable PYTHONLAZYIMPORTSALL is set
+    def check_env_var(self):
+        env_variables = os.environ
+        if "PYTHONLAZYIMPORTSALL" not in env_variables:
+            return False
+
+        # check this setting of this env_var
+        if env_variables["PYTHONLAZYIMPORTSALL"] == "0":
+            return False
+        else:
+            return True
+
+    # check if -L flag is added to the Python interpreter
+    def check_flag(self):
+        argvs = sys.argv
+        for argv in argvs:
+            if argv == "-L" or argv == "-l":
+                return True
+        return False
+
+    def is_enabled(self):
+        return self.enabled
+
+
+class TestEagerlyImport(unittest.TestCase):
+
+    def __init__(self, *args, **kwargs):
+        super(TestEagerlyImport, self).__init__(*args, **kwargs)
+        self.modules = sys.modules
+
+    def test_json(self):
+        # we did not import json, so we should not have these sub-modules
+        self.assertFalse("json.encoder" in self.modules)
+        self.assertFalse("json.decoder" in self.modules)
+
+    def test_sqlite3(self):
+        # we import sqlite3 at the beginning, so we should have sqlite3.dbapi2 in sys.modules
+        self.assertTrue("sqlite3.dbapi2" in self.modules)
```

Comment:
While checking for keys in `sys.modules` may look like a viable solution in this isolated test, when running multiple tests this could result in flaky results. e.gr. Think for a moment some other test in the test suite imports (and uses) `json` or `sqlite3` during its tests, and that such test runs before our own test. In this cases the test would have failed.

What could be other way of testing this while ensuring this doesn't happen?

Reply:
Is it better that we create two unique modules by ourselves and then replace `json` and `sqlite3` with them?
For example, we can create a module `Apple`, and a `Sub-Apple` is imported in `Apple`. Then, we could treat `Apple` as `json` and treat `Sub-Apple` as `json.encoder`. Does this make sense?

Reply:
I created `imported_module` and `imported_module2` to replace `sqlite3`. Also, I imported them by using different ways. For `json`, I created `non_imported_module` to replace its original position.